packages/client/src/components/Game/utils/level.ts

```diff
@@ -59,20 +96,218 @@ export class Level {
   }
 
   update(activeControlKeys: Set<CONTROL_KEYS>, lastControlKey: LastControlKey) {
-    this.player.update(activeControlKeys);
+    const isNotColiding = !this.canColide(this.player, lastControlKey);
+    this.player.update(activeControlKeys, isNotColiding);
     this.projectiles.forEach((projectile) => projectile.update());
     if (activeControlKeys.has(SPECIAL_CONTROL_KEYS.SPACE)) {
       const { x, y } = this.player.position;
       const velocity = getVelocity(lastControlKey);
 
       if (velocity) {
-        this.projectiles.push(new Projectile(x + CELL_SIZE / 2, y + CELL_SIZE / 2, velocity));
+        this.projectiles.push(new Projectile(x + TANK_SIZE / 2, y + TANK_SIZE / 2, velocity));
       }
 
       activeControlKeys.delete(SPECIAL_CONTROL_KEYS.SPACE);
     }
   }
 
+  canColide(tank: Tank, lastControlKey: LastControlKey) {
+    const { lastKey } = lastControlKey;
+    if (!lastKey) {
+      return;
```

Comment:
мне показалось что будет понятнее когда метод с приставкой can возвращает boolean а не boolean | undefined

```suggestion
      return false;
```

Reply:
Поправил